File: clojure/grade-school/src/grade_school.clj
(ns grade-school)

(defn grade [school grade]
  (get school grade []))

(defn add [school name grade]
  (update school grade (fnil conj []) name))

(defn sorted [school]
  (into (sorted-map) (map #(update % 1 sort) school)))

File: ruby/grade-school/grade_school.rb
class School
  attr_reader :grade_students

  def students(grade)
    @grade_students[grade]
  end

  def add(student, grade)
    @grade_students[grade] <<= student
    @grade_students[grade].sort!
  end

  def students_by_grade
    grade_students.sort_by(&:itself).map { { grade: _1, students: _2 } }
  end

  private

  def initialize
    @grade_students = Hash.new { [] }
  end
end
